tests/sentry/lang/native/test_symbolicator.py

```diff
@@ -103,3 +110,129 @@ def test_sources_custom_disabled(default_project):
+            },
+            {
+                "source": "sentry:microsoft",
+                "location": "http://microsoft.com/prefix/path1",
+                "download": {"status": "notfound"},
+            },
+        ]
+        response = {"modules": [{"debug_id": debug_id, "candidates": copy.copy(candidates)}]}
+        redacted = redact_internal_sources(response)
+        expected = [{"source": "sentry:microsoft", "download": {"status": "notfound"}}]
+        assert redacted["modules"][0]["candidates"] == expected
+
+    def test_notfound_omitted(self):
+        debug_id = "451a38b5-0679-79d2-0738-22a5ceb24c4b"
+        candidates = [
+            {
+                "source": "sentry:microsoft",
+                "location": "http://microsoft.com/prefix/path0",
+                "download": {"status": "notfound"},
+            },
+            {
+                "source": "sentry:microsoft",
+                "location": "http://microsoft.com/prefix/path1",
+                "download": {"status": "ok"},
+            },
+        ]
+        response = {"modules": [{"debug_id": debug_id, "candidates": copy.copy(candidates)}]}
+        redacted = redact_internal_sources(response)
+        expected = [{"source": "sentry:microsoft", "download": {"status": "ok"}}]
+        assert redacted["modules"][0]["candidates"] == expected
+
+    def test_multiple_notfound_filtered(self):
+        debug_id = "451a38b5-0679-79d2-0738-22a5ceb24c4b"
+        candidates = [
+            {
+                "source": "sentry:microsoft",
+                "location": "http://microsoft.com/prefix/path0",
+                "download": {"status": "notfound"},
+            },
+            {
+                "source": "sentry:microsoft",
+                "location": "http://microsoft.com/prefix/path1",
+                "download": {"status": "ok"},
+            },
+            {
+                "source": "sentry:apple",
+                "location": "http://microsoft.com/prefix/path0",
+                "download": {"status": "notfound"},
+            },
+            {
+                "source": "sentry:apple",
+                "location": "http://microsoft.com/prefix/path1",
+                "download": {"status": "ok"},
+            },
+        ]
+        response = {"modules": [{"debug_id": debug_id, "candidates": copy.copy(candidates)}]}
+        redacted = redact_internal_sources(response)
+        expected = [
+            {"source": "sentry:microsoft", "download": {"status": "ok"}},
+            {"source": "sentry:apple", "download": {"status": "ok"}},
+        ]
+        assert redacted["modules"][0]["candidates"] == expected
+
+    @pytest.mark.django_db
+    def test_sentry_project(self):
+        project = types.SimpleNamespace()
+        project.slug = "mystery"
+        project.organization = types.SimpleNamespace()
+        project.organization.slug = "underground"
+        project_source = get_internal_source(project)
```

Comment:
This seems rather sub-optimal, it doesn't really matter for what's being tested here but perhaps there exists a better mechanism I'm not aware of yet?

Reply:
I mean you can add an extra assertion here to make sure the test is easier to debug but I doubt this is going to matter for test perf, for example